lib/indacoin.ex

```diff
@@ -358,7 +358,11 @@ defmodule Indacoin do
 
   # NOTE: Indacoin can be really slow... we have to specify big timeout value
   defp do_request(method, url, body \\ "", headers \\ [], recv_timeout \\ 20_000) do
-    headers = Enum.into(headers, [{"Content-Type", "application/json"}])
+    headers =
+      if Enum.empty?(headers),
+        do: [{"Content-Type", "application/json"}],
+        else: headers ++ [{"Content-Type", "application/json"}]
```

Comment:
I think, the condition is not sufficient here and can be removed.

```elixir
headers ++ [{"Content-Type", "application/json"}]
```

will work just fine.